heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerMain.java

```diff
@@ -303,7 +303,7 @@ public boolean manageTopology() {
       // TODO(mfu): timeout should read from config
       SchedulerStateManagerAdaptor adaptor = new SchedulerStateManagerAdaptor(statemgr, 5000);
 
-      boolean isValid = validateRuntimeManage(adaptor, topologyName);
+      boolean isValid = (Command.KILL.equals(command)) || validateRuntimeManage(adaptor, topologyName);
```

Comment:
I think this is fine for KILL. It should be a blunt tool that cleans out any possible state it finds of the topology. It shouldn't fail when one part is missing and then leave behind another.